### PR TITLE
Migrate ESLint to 'flat' configuration

### PR DESCRIPTION
**Pull request by jonkoops**
Migrates the ESLint configuration to the [new 'flat' format](https://eslint.org/docs/latest/use/configure/migration-guide#start-using-flat-config-files), which will help facilitate the upgrade to the [recently released](https://eslint.org/blog/2024/04/eslint-v9.0.0-released/) ESLint 9.

### CODE REVIEW

File: js/.eslintrc.cjs [hunk 1]
```diff
@@ -1,113 +0,0 @@
...
-    extraFileExtensions: [".mjs"],
-  },
-  env: {
-    node: true,
-  },
-  plugins: ["lodash"],
-  extends: [
-    "eslint:recommended",
-    "plugin:import/recommended",
-    "plugin:import/typescript",
```
Comment (jonkoops) [inline js/.eslintrc.cjs:26]: I had to remove the import plugin due to incompatibility with the new 'flat' format (see https://github.com/import-js/eslint-plugin-import/issues/2556). Thankfully, we're not really using this plugin, so we can re-add it later when support eventually lands.

File: js/CODING_GUIDELINES.md
Comment (jonkoops) [inline js/CODING_GUIDELINES.md:1]: I've taken a moment to update the guidelines here, as they were no longer consistent with the code.

**Review by agagancarczyk — APPROVED** [review]

**Review by hmlnarik — APPROVED** [review]
LGTM

**Comment by melroy89** [comment]
Could you update README with some example usage of `eslint.config.js`?

**Comment by jonkoops** [comment]
> Could you update README with some example usage of `eslint.config.js`?

No, because this all works the same as it did before. Not sure what else one would do aside from run the linter, which is already documented.

**Pull request merged**